src/worker/helpers.go

```diff
@@ -166,46 +169,262 @@ func initOLDir(olPath string, dockerBaseImage string, newBase bool) (err error)
+				// Print an error if unmounting fails.
+				fmt.Printf("Could not unmount: %s\n", err.Error())
+				sandboxErrorCount += 1
+			}
+			if err := syscall.Rmdir(path); err != nil {
+				// Print an error if removing the mount directory fails.
+				fmt.Printf("Could not remove mount dir: %s\n", err.Error())
+				sandboxErrorCount += 1
+			}
+		}
+	}
+
+	// If we encountered any error while cleaning up the CGroup or the sandboxes
+	// return an error
+	if cgroupErrorCount != 0 || sandboxErrorCount != 0 {
+		if cgroupErrorCount != 0 {
+			fmt.Printf("%d error(s) while cleaning up cgroup.\n", cgroupErrorCount)
+		}
+		if sandboxErrorCount != 0 {
+			fmt.Printf("%d error(s) while cleaning up sandboxes.\n", sandboxErrorCount)
+		}
+		fmt.Printf("You can try to rerun the cleanup process again later.\n")
+		return fmt.Errorf("%d error(s) while cleaning up cgroup and %d error(s) while cleaning up sandbox", cgroupErrorCount, sandboxErrorCount)
+	}
+
+	// Attempt to unmount the main mount directory
+	fmt.Printf("Attempting to clean up main mount directory at %s\n", dirName)
+	if err := syscall.Unmount(dirName, syscall.MNT_DETACH); err != nil {
+		// Log an error if unmounting the main directory fails.
+		if errors.Is(err, syscall.EINVAL) {
+			fmt.Printf("Sandbox mount root is not mounted. No need to clean up.\n")
+		} else {
+			return fmt.Errorf("could not unmount %s: %s", dirName, err.Error())
+		}
+	}
+
+	// Remove the worker.pid file
+	if err := os.Remove(filepath.Join(olPath, "worker", "worker.pid")); err != nil {
+		// Return an error if removing worker.pid fails.
+		return fmt.Errorf("could not remove worker.pid: %s", err.Error())
+	}
+
+	return nil
+}
+
+// bringToStoppedClean tries the best to bring the state of OpenLambda to StoppedClean no mater which state it is in.
+func bringToStoppedClean(olPath string) error {
+	state, err := checkState()
+	if err != nil {
+		return fmt.Errorf("failed to check OL state: %s", err)
+	}
+
+	switch state {
+	case Running:
+		fmt.Println("An OpenLambda instance is currently running. Attempting to stop it...")
+		err := runningToStoppedClean()
+		if err != nil {
+			return fmt.Errorf("failed to stop the running OL instance: %s", err)
+		}
+		fmt.Println("Successfully stopped the running OpenLambda instance.")
+	case StoppedDirty:
+		fmt.Println("The previous OpenLambda instance did not exit cleanly. Attempting to clean up...")
+		err := stoppedDirtyToStoppedClean(olPath)
+		if err != nil {
+			return fmt.Errorf("failed to cleanup dirty shutdown: %s", err)
+		}
+		fmt.Println("Successfully cleaned up from the dirty shutdown.")
+	case StoppedClean:
+		fmt.Println("No OpenLambda instance is running. No further actions are needed.")
+	case Uninitialized:
+		fmt.Println("OpenLambda is not initialized. You should initialized it.")
+		return fmt.Errorf("cannot bring Uninitialized to StoppedClean")
+	default:
+		return fmt.Errorf("unrecognized state")
+	}
+
```

Comment:
A good cautious thing to do would be to call checkState again and make sure we actually are in StoppedClean now.  We used to have a lot of bugs where a clean shutdown wouldn't actually cleanup everything, so I'm always thinking about how we'll just similar bugs if they arise in the future.